### ISSUE TITLE
About metric caching

### ISSUE THREAD

**Issue by yihongXU** [vita-epfl/RAP#4]
Hello,

Very nice work! Congrats!

I am trying to extract the data but I have a question about the metric caching part:
# train metric caching
python navsim/planning/script/run_training_metric_caching.py \
train_test_split=navtrain \
cache.cache_path=./train_metric_cache \
worker.threads_per_node=32 \

If I understand well, the metric caching trying to save pdm trajectories for training purposes. Therefore, all trajectories from all agents should be cached. Since you augment the dataset not only for the ego traj. but other agents'. 

I want to know where you modified the metric caching to make sure that you cache all pdm traj. for all agents?

best,

**Comment by Alan-LanFeng** [comment]
We only cached the metric for ego trajs. It would be difficult to get other agents’ metrics because of the lack of annotations. During training, We only calculate the score loss for ego trajs.

Please see here: https://github.com/vita-epfl/RAP/blob/1d9a886e00a202021e0378b96f19438035cec75d/navsim/planning/script/run_training.py#L149
